independent-projects/qute/core/src/main/java/io/quarkus/qute/Parser.java

```diff
@@ -331,13 +337,23 @@ private void flushTag() {
             typeInfos.put(key, "[" + value + "]");
 
         } else {
-            sectionBlockStack.peek()
-                    .addNode(new ExpressionNode(apply(content), engine, new OriginImpl(line, templateId, variant)));
+            sectionBlockStack.peek().addNode(new ExpressionNode(apply(content), engine, origin()));
         }
         this.buffer = new StringBuilder();
     }
 
-    private void processParams(String label, Iterator<String> iter) {
+    private void parserError(String message) {
+        StringBuilder builder = new StringBuilder("Parser error");
+        if (!id.equals(generatedId)) {
+            builder.append(" in template [").append(id).append("]");
+        }
+        builder.append(" on line ").append(line).append(": ")
+                .append(message);
+        throw new TemplateException(origin(),
```

Comment:
I think you shouldn't throw the exception in the method. You should return it and throw it in the caller.

This way, IDEs can analyze properly the flow of the methods.

Reply:
Ok, I can modify the method. No problem.

Reply:
Cool thanks. I already was bitten by this, totally missing a coding error because the IDE was unable to tell me some part of the code was unreachable.